### ISSUE TITLE
Environment variables are named with `5`

### ISSUE THREAD

**Issue by leonerd** [Perl/perl5#21056]
(Relating to #21055)

Many (all?) of the environment variables that `perl` looks in or uses for adjusting its behaviour are all named `PERL5...`. This encodes the major version number 5 into their name, and makes it harder increase that number in future.

In particular, if we were to bump the major version to 7, then it might seem weird to have to

```
$ PERL5OPT=-MFoo PERL5LIB=/home/whereever/perl7 perl ...
```

**Comment by xenu** [comment]
Removing `PERL5FOO` would be a major breakage for little gain, while having both `PERLFOO` and `PERL5FOO` would introduce even more confusion.

IMO that '5' in the names isn't a big deal. We should keep it and document that those variables affect all Perl versions.

**Comment by leonerd** [comment]
Yeah; that may be a decision we take. It's a bit ugly but ... eh. It doesn't really *break* anything as such, so perhaps it's fine to leave them. But still we should come to a decision and document what we decided and why.

**Comment by haarg** [comment]
`PERLLIB` already exists.

**Comment by rwp0** [comment]
Creating `PERLOPT` etc, and making `PERL5OPT` a compatibility alias to it sounds good to me.

The same goes for everything else.

**Comment by rwp0** [comment]
Although not exactly much related, created:

- https://github.com/Perl/perl5/issues/21078

**Comment by argrath** [comment]
FYI: [PPC](https://github.com/Perl/PPCs/blob/main/ppcs/ppc0025-perl-version.md) says:

---

A number of new environment variables are introduced which supersede existing environment variables if present:

* `PERL5LIB` → `PERL_LIB` (but cf. [Open Issues](https://github.com/Perl/PPCs/blob/main/ppcs/ppc0025-perl-version.md#open-issues) regarding this one)
* `PERL5OPT` → `PERL_OPT`
* `PERL5DB` → `PERL_DB`
* `PERL5DB_THREADED` → `PERL_DB_THREADED`
* `PERL5SHELL` → `PERL_SHELL`

As a result, all Perl environment variables except `PERLIO` now follow the pattern `PERL_*`.

**Comment by rwp0** [comment]
> * PERL_DB_THREADED

Yeah, that's even better in being more readable.
